nomad/structs/volumes.go

```diff
@@ -100,12 +100,10 @@ func (v *VolumeRequest) Copy() *VolumeRequest {
 	nv := new(VolumeRequest)
 	*nv = *v
 
-	if v.MountOptions == nil {
-		return nv
+	if v.MountOptions != nil {
```

Comment:
Most of our `Copy` functions have a `nil` check built in making this redundant 
https://github.com/hashicorp/nomad/blob/master/nomad/structs/csi.go#L151

Reply:
Yeah, I debated that myself - in my earlier suggestion above, I didn't include the conditional.  However, I think having the explicit conditional is better for readability, as it's obvious for the reader that there is no NPE bug without needing to jump to the Copy definition.